kong/plugins/aws-lambda/schema.lua

```diff
@@ -58,6 +58,7 @@ return {
         } },
         { host = typedefs.host },
         { port = typedefs.port { default = 443 }, },
+        { https = { type = "boolean", default = true }, },
```

Comment:
I recommend reconsidering the true by default design here and instead renaming the field to `disable_https` with a default value of `false`.
If someone is running a version of Kong before this patch, and upgrades, `config.https` will not be present in the database (since there are no migrations) and this value will be `null`(?). This is generally not a problem in DB-less mode where missing defaults are injected but the same cannot be said with confidence about Kong Manager interactions and traditional mode deployments.
I also generally discourage patches that add 'true by default' design since they add noise to the configuration for the users. More so for configuration options that aren't going to be used much and are a security risk if disabled.

What do you think?

Reply:
Initially, I chose the way to add `https` field because I want to keep a similar schema field like [the Azure serverless function plugin](https://docs.konghq.com/hub/kong-inc/azure-functions/).

> This is generally not a problem in DB-less mode where missing defaults are injected but the same cannot be said with confidence about Kong Manager interactions and traditional mode deployments.

I think this would generally also not be a problem in Kong Manager and traditional mode. Such new schema fields are not real database columns, they're only JSON fields inside the database and if they did not exist before the upgrade, and once upgraded, when the plugin entity is selected from the database, the missing fields will be filled automatically with the default value(I think `handle_missing_field` is doing the job here)

Reply:
Discussed offline, and to avoid further confusion I renamed the field to `disable_https`.

Please note that changing this field does not change port number and currently, we do not have a proper & friendly way to dynamically set port value according to the https related field. So I'll add some notice to the document and we may consider simplifying the endpoint-related configs(host, port, scheme, etc.) to one field `endpoint` in the future.